wordpress-mobile/WordPress-Android
# Use View Binding in Reader ViewHolders

This PR introduces View Bindings for the Reader ViewHolders. The approach is a bit similar to #14211, except it delegates the layout inflation to an extension function (see 91bc58f). I believe this makes abstract ViewHolders like the `ReaderViewHolder` [a lot cleaner](https://github.com/wordpress-mobile/WordPress-Android/pull/14225/files#diff-cbef3240e5a7f026946cea2fb45debec28306677afb06c2321063c8e8ac1d1f7) and doesn't add much more complexity to its subclasses or concrete implementations like the [`ReaderInterestViewHolder`](https://github.com/wordpress-mobile/WordPress-Android/pull/14225/files#diff-e7ea8e780247dda3e6c846155f2fb7aadb3eaa281ac5c82b7eba4e6c89679e15).

To test:

Smoke test the Reader, with special attention for the Discover section and the Interests screen, and make sure everything is working as expected and nothing looks out of place.

PR submission checklist:
